src/main/java/de/hysky/skyblocker/skyblock/item/tooltip/adders/BitsHelper.java

```diff
@@ -0,0 +1,448 @@
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class BitsHelper extends SimpleContainerSolver implements TooltipAdder {
+    private static final Pattern BITS_PATTERN = Pattern.compile("Cost (?<amount>[\\d,]+) Bits");
+    private static final Pattern CATEGORY_PATTERN = Pattern.compile("Click to browse!");
+    @Language("RegExp") private static final String TITLE_PATTERN = ".*(?:Community Shop|Bits Shop).*";
+    private static final NumberFormat DECIMAL_FORMAT = NumberFormat.getInstance(Locale.US);
+    private static final Logger LOGGER = LoggerFactory.getLogger("Skyblocker Bits");
+    private static final String LOGS_PREFIX = "[Skyblocker Bits Helper] ";
+
+    public static final BitsHelper INSTANCE = new BitsHelper();
+
+    public BitsHelper() {
+        super(TITLE_PATTERN);
+    }
+
+    private Map<String, Pair<String, Long>> categoryOutput = new HashMap<>();
+    private int bestSlotIndexSelling = -1;
+    private int bestSlotIndexAll = -1;
+
+    @Override
+    public List<ColorHighlight> getColors(Int2ObjectMap<ItemStack> slots) {
+        List<ColorHighlight> highlights = new ArrayList<>();
+        BestItemsResult bestItemsResult = calculateBestItems(slots);
+
+        bestSlotIndexSelling = bestItemsResult.bestSlotIndexSelling;
+        bestSlotIndexAll = bestItemsResult.bestSlotIndexAll;
+
+        // If best selling = best overall, only green will appear. This is intended
+        if (bestSlotIndexSelling != -1 && bestSlotIndexSelling == bestSlotIndexAll) {
+            highlights.add(ColorHighlight.green(bestSlotIndexSelling));
+        } else {
+            if (bestSlotIndexSelling != -1) {
+                highlights.add(ColorHighlight.green(bestSlotIndexSelling));
+            }
+            if (bestSlotIndexAll != -1) {
+                highlights.add(ColorHighlight.yellow(bestSlotIndexAll));
+            }
+        }
+        return highlights;
+    }
+
+    /**
+     * Why there is no simpler way to get those?
+     */
+    private boolean megamindInLogs = false;
+    private Int2ObjectMap<ItemStack> getSlots() {
+        MinecraftClient client = MinecraftClient.getInstance();
+        if (client.currentScreen instanceof HandledScreen) {
+            HandledScreen<?> screen = (HandledScreen<?>) client.currentScreen;
+            ScreenHandler handler = screen.getScreenHandler();
+
+            Int2ObjectMap<ItemStack> slots = new Int2ObjectOpenHashMap<>();
+            for (int i = 0; i < handler.slots.size(); i++) {
+                slots.put(i, handler.getSlot(i).getStack());
+            }
+            if (!megamindInLogs) {
+                LOGGER.info(LOGS_PREFIX+"No slots?\n" +
+                            "⠀⣞⢽⢪⢣⢣⢣⢫⡺⡵⣝⡮⣗⢷⢽⢽⢽⣮⡷⡽⣜⣜⢮⢺⣜⢷⢽⢝⡽⣝\n" +
+                            "⠸⡸⠜⠕⠕⠁⢁⢇⢏⢽⢺⣪⡳⡝⣎⣏⢯⢞⡿⣟⣷⣳⢯⡷⣽⢽⢯⣳⣫⠇\n" +
+                            "⠀⠀⢀⢀⢄⢬⢪⡪⡎⣆⡈⠚⠜⠕⠇⠗⠝⢕⢯⢫⣞⣯⣿⣻⡽⣏⢗⣗⠏⠀\n" +
+                            "⠀⠪⡪⡪⣪⢪⢺⢸⢢⢓⢆⢤⢀⠀⠀⠀⠀⠈⢊⢞⡾⣿⡯⣏⢮⠷⠁⠀⠀\n" +
```

Comment:
Might wanna add a space before the \n so the adders are aligned, also you might be able to use a text block instead of having to do the \n and concats